BridgeBot/bidding.py

```diff
@@ -1,47 +1,53 @@
-from BridgeBot.enums import players, strains, contracts
+from enums import Players, Strains, AuctionStatus, Doubles, Contracts, PASS, Team
 
-doubles = [None, 'X', 'XX']
 
-class Auction:
-    contract = {'level': 0,
-                'strain': None,
-                'doubled': None,
-                'declarer': None}
+class InvalidDealerException(Exception):
+    pass
+
 
+class Contract:
+    def __init__(self):
+        self.level = 0
+        self.strain = Strains.NT
+        self.doubled = Doubles.NONE
+        self.declarer = None
+
+class Auction:
     def __init__(self, dealer):
         self.dealer = dealer.upper()
 
-        if not dealer in players:
-            raise Exception('Invalid dealer.')
+        if not isinstance(dealer, Players):
+            raise InvalidDealerException('Invalid dealer')
 
         self.record = {
-            'NORTH': [],
-            'EAST': [],
-            'SOUTH': [],
-            'WEST': []
+            Players.NORTH: [],
+            Players.EAST: [],
+            Players.SOUTH: [],
+            Players.WEST: []
         }
 
         self.ns_first_bid = {
-            'CLUBS':"",
-            'DIAMONDS':"",
-            'HEARTS':"",
-            'SPADES':"",
-            'NT':""
+            Strains.CLUBS: None,
+            Strains.DIAMONDS: None,
+            Strains.HEARTS: None,
+            Strains.SPADES: None,
+            Strains.NT: None
         }
 
         self.ew_first_bid = {
-            'CLUBS': "",
-            'DIAMONDS': "",
-            'HEARTS': "",
-            'SPADES': "",
-            'NT': ""
+            Strains.CLUBS: None,
+            Strains.DIAMONDS: None,
+            Strains.HEARTS: None,
+            Strains.SPADES: None,
+            Strains.NT: None
         }
 
-        self.player_index = players.index(self.dealer)
+        self.player_index = Players.players().index(self.dealer)
         self.last_bidder_index = None
 
-        for i in range(self.player_index):
-            self.record[players[i]].append("-")
+        # Not sure about the following line
```

Comment:
This is for human readability. Say you're playing north, but south bids first. It's easier to display

```
NORTH   EAST    SOUTH   WEST
  -       -       P      1S   
  P      2C       X

Without the placeholders, it's difficult to see who started (or even to keep the rounds separate)
```